tests/auth.test.ts

```diff
@@ -1,24 +1,27 @@
 import { expect, test, type Page } from "@playwright/test";
 
-test.describe("Auth", () => {
+test.describe.serial("Auth", () => {
   let page: Page;
   test.beforeAll(async ({ browser }) => {
     page = await browser.newPage();
     await page.goto("/auth-test");
   });
+  test.afterAll(async () => {
+    await page.close();
+  });
 
   test("Renders UI conditionally based on auth state", async () => {
-    expect(page.getByText("Signed In")).not.toBeVisible();
-    expect(page.getByText("Signed Out")).toBeVisible();
+    await expect(page.getByText("Signed Out")).toBeVisible();
+    await expect(page.getByText("Signed In")).toBeHidden();
   });
 
   test("User can sign in and out", async () => {
-    await page.click("text=Sign In");
-    await page.waitForSelector("text=Sign Out");
+    await page.getByRole("button", { name: "Sign In" }).click();
```

Comment:
We recommend using role selectors, which do access-ability testing as a drive-by. All our assertions wait automatically, so you don't need waitForSelector. By using expect, you'll also get good error reporting.